### ISSUE TITLE
Have a small issue with speed

### ISSUE THREAD

**Issue by marinosmarkitsis** [bmaltais/kohya_ss#3504]
5080+32gb ram

{
"LoRA_type": "LyCORIS/LoCon",
"network_module": "lycoris.kohya",
"dora_wd": true,
"output_name": "bbfamiliar_v4_1024sq",
"optimizer": "Prodigy",
"optimizer_args": "decouple=True weight_decay=0.01 d_coef=2.0",
"lr_scheduler": "cosine_with_restarts",
"lr_warmup_steps": 100,
"learning_rate": 1.0,
"unet_lr": 1.0,
"text_encoder_lr": 1,
"train_batch_size": 2,
"gradient_accumulation_steps": 1,
"epoch": 8,
"save_every_n_epochs": 2,
"mixed_precision": "bf16",
"full_bf16": true,
"save_precision": "bf16",
"network_dim": 256,
"network_alpha": 128,
"conv_dim": 64,
"conv_alpha": 32,
"gradient_checkpointing": true,
"cache_latents": true,
"cache_latents_to_disk": false,
"enable_bucket": false,
"bucket_resolution_steps": 64,
"min_bucket_resolution": 1024,
"max_bucket_resolution": 1024,
"resolution": "1024,1024",
"noise_offset": 0.05,
"caption_extension": ".txt",
"shuffle_caption": true,
"caption_dropout_rate": 0.05
}

<img width="1600" height="1200" alt="Image" src="https://github.com/user-attachments/assets/2a28da66-f1d4-4b65-81f5-73f1a3fa8163" />

<img width="1600" height="901" alt="Image" src="https://github.com/user-attachments/assets/5d62c0d2-4ebe-489d-8470-9d66de2614df" />

**Comment by bruvic62** [comment]
THE INTEL/UBUNTU AI MANIFESTO: BEYOND THE NVIDIA BARRIER
The Headache: The Windows/CUDA Deadlock
Most AI training software is built as an Nvidia-first "hostage situation." On Windows, your system fights a two-front war: the OS hogs vital Intel resources, and the software throws "Copy Fail" logic flaws because it’s searching for an Nvidia card that doesn't exist. This leads to the "Juggernaut Clog"—system freezes, thermal throttling, and a massive performance drain that makes laptop training feel impossible.

The Pivot: Re-Engineering the Flow on Ubuntu
To win, we stopped trying to "fix" Windows and moved to a native Ubuntu/Linux environment. We stopped faking hardware and started re-engineering the software bridge. By "boring out the pipes," we moved from a stuttering crawl to a 7x speed increase. We aren't making a laptop act like a server; we are simply allowing the Intel hardware to finally run at 100% efficiency.

THE TECHNICAL STACK (CONFIRMED WORKING)
OS: Ubuntu / Linux

Kernel Fix: Python 3.10 / PyTorch 2.6.0+cpu / IPEX 2.6.0

Hardware: 12th Gen Intel Core i5-1235U (10 Physical / 12 Logical Cores)

The Accelerator: 12 CPU threads / FP16 Mixed Precision

THE "PHANTOM LIMB" FIXES (HOW WE DID IT)
If you are technical, you know the software "expects" CUDA. You don't delete CUDA; you neutralize it so the CPU can take over the math.

1. The NMS Surgery (The "Eureka" Moment)
The torchvision::nms operator is hard-coded for Nvidia. When it doesn't see a GPU, it panics and crashes.

The Fix: Manually wrap @torch._custom_ops.impl_abstract("torchvision::nms") and def meta_nms in a try/except RuntimeError block within _meta_registrations.py. This forces the system to fall back to the CPU logic it was previously ignoring.

2. The patchelf Security Clearance
Linux security often blocks Intel performance libraries (libintel-ext-pt-cpu.so) by default.

The Fix: Use patchelf 0.18.0 to clear the execstack flag. This is a "silent killer" that stops training without an error log; clearing it unlocks the Intel "muscles".

3. Removing the "Poison" (xformers)

The Action: Completely uninstall xformers. It is a hard-coded Nvidia dependency that acts as a data clog on Intel setups. Stick to CrossAttention: None and IPEX optimization.

THE "NO-FLY" ZONE (DO NOT DO THIS)
NO AdamW8bit: This optimizer is a CUDA-only trap that will crash your engine. Stick to AdamW or a CPU-optimized equivalent.

NO Windows Backgrounds: Ditch the Windows "handbrake". The background bloat and driver conflicts are what actually kill your speed.

NO SSL Guessing: Manually resolve certificate errors to ensure the Intel Extension for PyTorch (IPEX) bridge is fully built.

THE FINAL RESULT
Performance: All 12 logical threads fully engaged with a 7x speed boost over Windows.

Thermal Stability: System stays between 48°C - 59°C instead of hitting 100°C thermal limits.

Efficiency: Native Linux execution removes the "Juggernaut" overhead.

The Verdict: This was a 10-hour technical siege involving 7 failed installation attempts. If your system "throws a code," it’s just a poorly documented dependency. Don't give up. The "pipes" can be bored out—you just have to be more stubborn than the code.

**Comment by bmaltais** [comment]
This is more a question for the sd-scripts upstream solution the GUI is wrapping.
